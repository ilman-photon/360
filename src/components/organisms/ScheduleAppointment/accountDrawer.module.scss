.listItemDrawer {
  border-bottom: black;
  border-width: 0;
  border-style: solid;
  border-color: rgba(0, 0, 0, 0.12);
  border-bottom-width: thin;
}

.formButton {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 13px 20px;
  border-radius: 30px;
  font-size: 14px;
  height: 40px;
  width: 100%;
  text-transform: none;

  &.outlined {
    background-color: transparent;
    border: solid 1px #000000;
    color: #000000;
    width: 80%;
  }
}

.logoutButton {
  width: 100%;
  justify-content: center;
}

///@at-root
