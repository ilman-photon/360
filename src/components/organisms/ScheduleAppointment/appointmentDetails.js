import * as React from "react";
import { Stack, Button } from "@mui/material";
import styles from "./Style.module.scss";
import { useTranslation } from "next-i18next";
import AccountCard from "../../molecules/AccountCard/accountCard";
import LabelWithInfo from "../../atoms/LabelWithInfo/labelWithInfo";

import EditOutlinedIcon from "@mui/icons-material/EditOutlined";
import CalendarTodayIcon from "@mui/icons-material/CalendarToday";
import ContactMailIcon from "@mui/icons-material/ContactMail";
import BusinessIcon from "@mui/icons-material/Business";

export default function AppointmentDetails() {
  const { t } = useTranslation("translation", {
    keyPrefix: "scheduleAppoinment",
  });

  return (
    <AccountCard
      title={t("appointmentDetails")}
      //   isEditing={isEditing}
      // OnEditClicked={OnEditClicked}
      actionContent={
        <Button
          // onClick={OnEditClicked}
          variant="text"
          className={styles.editButton}
        >
          <EditOutlinedIcon sx={{ width: 20, height: 20 }} />
          <div type="link" style={{ marginLeft: 4, color: "#008294" }}>
            Edit
          </div>
        </Button>
      }
    >
      <Stack spacing={2}>
        <LabelWithInfo label="Name" titleIcon={<CalendarTodayIcon />}>
          Saturday, Sep 11 - 8:30 am EST
        </LabelWithInfo>

        <LabelWithInfo label="Preferred Name" titleIcon={<ContactMailIcon />}>
          No Insurance provided
        </LabelWithInfo>

        <LabelWithInfo label="fsd Name" titleIcon={<BusinessIcon />}>
          Eye Exam
        </LabelWithInfo>
      </Stack>
    </AccountCard>
  );
}
