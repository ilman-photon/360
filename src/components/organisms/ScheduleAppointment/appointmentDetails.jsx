import * as React from "react";
import { Stack, Button, Typography, Box } from "@mui/material";
import { useTranslation } from "next-i18next";
import AccountCard from "../../molecules/AccountCard/accountCard";
import LabelWithInfo from "../../atoms/LabelWithInfo/labelWithInfo";

import EditOutlinedIcon from "@mui/icons-material/EditOutlined";
import EyeIcon from "../../../assets/icons/EyeIcon";

import { styles } from "./style";
import { formatAppointmentDate } from "../../../utils/dateFormatter";
import { colors } from "../../../styles/theme";
import { TEST_ID } from "../../../utils/constants";
import Image from "next/image";

export default function AppointmentDetails({
  appointmentData = {},
  OnEditClicked = () => {
    // This is intended
  },
}) {
  const { t } = useTranslation("translation", {
    keyPrefix: "scheduleAppoinment",
  });

  return (
    <Box>
      <AccountCard
        title={t("appointmentDetails")}
        isAppoinment={true}
        sx={{
          ".MuiCardContent-root": {
            p: 3,
          },
        }}
        actionContent={
          <Button
            data-testid={
              TEST_ID.SCHEDULE_APPOINTMENT_TEST_ID.APPOINTMENT_DETAILS
                .editButton
            }
            variant="text"
            className={styles.editButton}
            onClick={OnEditClicked}
            aria-label="Edit"
          >
            <EditOutlinedIcon
              sx={{ width: 20, height: 20, color: "#008294" }}
              aria-hidden={"false"}
            />
            <div type="link" style={styles.editLink}>
              Edit
            </div>
          </Button>
        }
      >
        <Stack spacing={2}>
          <LabelWithInfo
            label="Date and time"
            iconWidth={"24px"}
            titleIcon={
              <Image
                width="20"
                height="20"
                src={"/icon-calendar.png"}
                alt="date and time"
                aria-label={"Calendar icon"}
                aria-hidden={"false"}
                sx={{ color: colors.darkGreen }}
              />
            }
            sxRow={{ justifyContent: "unset" }}
            sxText={{
              height: "26px",
              "font-style": "normal",
              "font-weight": "600",
              "font-size": "18px",
              "line-height": "26px",
              paddingLeft: "4px",
              "letter-spacing": "0.0016em",
              color: colors.darkGreen,
              fontWeight: "600",
              fontSize: "16px",
            }}
          >
            <Typography
              variant="bodyMedium"
              sx={{
                color: "#003B4A",
                fontFamily: "Museo Sans",
                fontSize: "16px",
                fontWeight: "400",
                "font-style": "normal",
                "font-weight": "400",
                "font-size": "16px",
                "line-height": "24px",
              }}
              tabIndex={"0"}
              data-testid={
                TEST_ID.SCHEDULE_APPOINTMENT_TEST_ID.APPOINTMENT_DETAILS.date
              }
            >
              {formatAppointmentDate(
                appointmentData.date,
                appointmentData.timeZone
              )}
            </Typography>
          </LabelWithInfo>
          <LabelWithInfo
            label="Insurance"
            iconWidth={"24px"}
            titleIcon={
              <Image
                width="24"
                height="24"
                src={"/icon-profile.png"}
                alt="insurance"
                aria-label={"Mail icon"}
                aria-hidden={"false"}
              />
            }
            sxRow={{ justifyContent: "unset" }}
            sxText={{
              paddingLeft: "4px",
              width: "85px",
              height: "26px",
              "font-style": "normal",
              "font-weight": "600",
              "font-size": "18px",
              "line-height": "26px",
              "letter-spacing": "0.0016em",
              color: colors.darkGreen,
              fontWeight: "600",
              fontSize: "16px",
            }}
          >
            <Typography
              variant="bodyMedium"
              sx={{
                color: "#003B4A",
                fontFamily: "Museo Sans",
                "font-style": "normal",
                "font-weight": "400",
                "font-size": "16px",
                "line-height": "24px",
              }}
              tabIndex={"0"}
            >
              {appointmentData.insuranceCarrier?.name
                ? appointmentData.insuranceCarrier?.name
                : "No Insurance provided"}
            </Typography>
          </LabelWithInfo>
          <LabelWithInfo
            label="Purpose of visit"
            iconWidth={"24px"}
            titleIcon={
              <EyeIcon />
              // <Image
              //   width="24"
              //   height="24"
              //   src={"/icon-eye-contacts.png"}
              //   alt="purpose of visit"
              //   aria-label={"Building icon"}
              //   aria-hidden={"false"}
              //   sx={{ color: colors.darkGreen }}
              // />
            }
            sxRow={{ justifyContent: "unset" }}
            sxText={{
              paddingLeft: "4px",
              height: "26px",
              "font-style": "normal",
              "font-weight": "600",
              "font-size": "18px",
              "line-height": "26px",
              "letter-spacing": "0.0016em",
              color: colors.darkGreen,
              fontWeight: "600",
              fontSize: "16px",
            }}
          >
            <Typography
              variant="bodyMedium"
              sx={{
                color: colors.darkGreen,
                fontFamily: "Museo Sans",
                "font-style": "normal",
                "font-weight": "400",
                "font-size": "16px",
                "line-height": "24px",
              }}
              tabIndex={"0"}
              aria-label={
                appointmentData.appointmentType ||
                "no purpose of visit provided"
              }
            >
              {appointmentData.appointmentType || "-"}
            </Typography>
          </LabelWithInfo>
        </Stack>
      </AccountCard>
    </Box>
  );
}
