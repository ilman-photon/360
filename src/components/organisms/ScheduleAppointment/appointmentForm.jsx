import * as React from "react";
import {
  Box,
  Stack,
  Typography,
  Button,
  Divider,
  useMediaQuery,
} from "@mui/material";
import { useTranslation } from "next-i18next";
import { styles } from "./style";
import { useForm, Controller, useFormState } from "react-hook-form";
import RowRadioButtonsGroup from "../../atoms/RowRadioButtonsGroup/rowRadioButtonsGroup";
import { StyledInput } from "../../atoms/Input/input";
import { Regex } from "../../../utils/regex";
import constants from "../../../utils/constants";
import Link from "@mui/material/Link";
import { getLinkAria } from "../../../utils/viewUtil";
import FormLabel from "@mui/material/FormLabel";
import { useRouter } from "next/router";
import FormMessage from "../../molecules/FormMessage/formMessage";
import { PasswordValidator } from "../../molecules/PasswordValidator/passwordValidator";
import { colors } from "../../../styles/theme";

const DisclaimerText = (data) => {
  return (
    <FormLabel
      id="row-input-disclaimer"
      sx={{
        fontSize: 12,
        "&.Mui-focused": {
          color: "black",
        },
        paddingLeft: 2,
        display: "inline-flex",
        alignItems: "center",
        color: "#424747",
        fontFamily: "Libre Franklin",
      }}
    >
      {data.label}
    </FormLabel>
  );
};

export const isDOB = (value) => {
  let date = new Date().getFullYear();

  if (value.getYear() < 0) {
    return false;
  }
  if (value.getFullYear() <= date) {
    return true;
  }
  return value.getMonth() <= 12;
};

export default function AppointmentForm({
  isForMyself,
  OnClickSchedule = () => {
    // This is intentional
  },
  patientData = {},
  OnSubmit = () => {
    // This is intended
  },
  OnClickSignIn = () => {
    // This is intended
  },
  formMessage = null,
}) {
  const { handleSubmit, control, watch, setValue } = useForm({
    defaultValues: patientData,
  });
  const [open, setOpen] = React.useState(false);

  const firstNameRef = React.useRef(null);
  const lastNameRef = React.useRef(null);
  const emailRef = React.useRef(null);
  const mobileRef = React.useRef(null);
  const dobRef = React.useRef(null);
  const { errors, isSubmitting } = useFormState({
    control,
  });

  React.useEffect(() => {
    console.log(errors);
    if (errors.firstName) {
      firstNameRef.current?.focus();
    } else if (errors.lastName) {
      lastNameRef.current?.focus();
    } else if (errors.email) {
      emailRef.current?.focus();
    } else if (errors.mobile) {
      mobileRef.current?.focus();
    } else if (errors.dob) {
      dobRef.current?.focus();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [isSubmitting]);

  const router = useRouter();

  const { SCHEDULE_GUEST_TEST_ID } = constants.TEST_ID;
  const validatePassword = (errors1 = [], errors2 = []) => {
    return errors1.length === 0 && errors2.length <= 1;
  };

  const onSubmit = (data) => {
    OnClickSchedule(data);
    OnSubmit(data);
  };

  const onLogin = () => {
    OnClickSignIn();
    router.push("/patient/login");
  };

  const options = [
    {
      label: "Email",
      value: "email",
      testId: SCHEDULE_GUEST_TEST_ID.emailradio,
    },
    {
      label: "Phone",
      value: "phone",
      testId: SCHEDULE_GUEST_TEST_ID.phoneradio,
    },
    { label: "Both", value: "both", testId: SCHEDULE_GUEST_TEST_ID.bothradio },
  ];

  const watchedPassword = watch("password", "");
  const [watchedEmail, watchedMobile, watchedPreferredCommunication] = watch([
    "email",
    "mobile",
    "preferredCommunication",
    "password",
  ]); // you can also target specific fields by their names
  const getRegisteredUsername = () => {
    return (
      watchedEmail || watchedMobile || "(auto-populated email id/phone number)"
    );
  };

  const passwordValidator = [
    {
      label: "Length: 8-20 characters",
      validate: !Regex.lengthRegex.test(watchedPassword),
      mandatory: true,
    },
    {
      label: "Contain at least 3 out of 4 types of characters below:",
      passesValidation: 3,
      text: true,
      children: [
        {
          label: "At least One Numeric",
          validate: Regex.numberRegex.test(watchedPassword),
        },
        {
          label: "At least One Upper case Alpha",
          validate: Regex.upperCaseRegex.test(watchedPassword),
        },
        {
          label: "At least One Lower case Alpha",
          validate: Regex.lowerCaseRegex.test(watchedPassword),
        },
        {
          label: "At least One Special character (no spaces)",
          validate: Regex.specialRegex.test(watchedPassword),
        },
      ],
    },
    {
      label: "Password should not contain your username",
      validate: watchedPassword?.indexOf(watchedEmail || watchedMobile) > -1,
      mandatory: true,
    },
    {
      label: "New password must not match current password",
      validate: false,
    },
  ];
  const isPasswordError = watchedPassword?.length > 0; // && passwordValidator.filter(v => v.validate).length > 0

  const { t, ready } = useTranslation("translation", {
    keyPrefix: "scheduleAppoinment",
  });

  const isDesktop = useMediaQuery("(min-width: 769px)");

  const is3of4 = (pass) => {
    let passes = 0;
    if (Regex.alphabethRegex.test(pass)) {
      ++passes;
    }
    if (Regex.specialRegex.test(pass)) {
      ++passes;
    }
    if (pass.indexOf(watchedEmail || watchedMobile) > -1) {
      ++passes;
    }
    if (!Regex.hasTripleRegex.test(pass)) {
      ++passes;
    }
    return passes >= 3 ? true : false;
  };

  const getTitleStyle = () => {
    if (isForMyself) {
      return isDesktop
        ? {
            fontSize: "32px",
            fontFamily: "Bw Nista Geometric DEMO",
            fontWeight: "500",
          }
        : {
            fontSize: "26px",
            fontFamily: "Bw Nista Geometric DEMO",
          };
    } else {
      return {
        fontSize: isDesktop ? "26px" : "22px",
        fontFamily: "Inter",
        fontWeight: "500",
      };
    }
  };

  const formMessageComp = React.useRef(null);
  React.useEffect(() => {
    if (formMessageComp.current)
      formMessageComp.current.scrollIntoView({
        behavior: "smooth",
        block: "end",
        inline: "nearest",
      });
  }, [formMessage]);

  React.useEffect(() => {
    const isMobileInputEmpty = watchedMobile == "(" || !watchedMobile;
    if (watchedEmail && isMobileInputEmpty) {
      setValue("preferredCommunication", "email");
    } else if (!watchedEmail && !isMobileInputEmpty) {
      setValue("preferredCommunication", "phone");
    } else {
      setValue("preferredCommunication", "both");
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [watchedEmail, watchedMobile]);

  return (
    <>
      {ready && (
        <Stack spacing={2}>
          {/* <Stack spacing={2}> */}
          <Box sx={{ m: 1 }}>
            <Typography
              sx={getTitleStyle()}
              tabIndex={0}
              aria-label={
                isForMyself
                  ? `${t("selfTitle")} title`
                  : `${t("someoneElseTitle")} title`
              }
              aria-roledescription=""
            >
              {isForMyself ? t("selfTitle") : t("someoneElseTitle")}
            </Typography>

            {formMessage?.content ? (
              <FormMessage
                ref={formMessageComp}
                success={formMessage.success}
                title={formMessage.title}
                isBackToLogin={formMessage.isBackToLogin}
                isSchedule
                sx={{
                  width: "71%",
                }}
              >
                {formMessage.content}
              </FormMessage>
            ) : (
              ""
            )}
            {isForMyself ? (
              <Box sx={{ mt: 2, display: "flex" }}>
                <Typography
                  sx={styles.signInInfoLabel}
                  tabIndex={0}
                  aria-label={`${t("sigInInfo")}`}
                  aria-roledescription=""
                  variant="h1"
                >
                  {t("sigInInfo")}
                </Typography>
                <Link
                  sx={styles.link}
                  data-testid={SCHEDULE_GUEST_TEST_ID.signInlink}
                  {...getLinkAria(t("login"))}
                  onClick={onLogin}
                >
                  {t("login")}
                </Link>
              </Box>
            ) : null}
          </Box>

          <form
            onSubmit={handleSubmit(onSubmit)}
            style={styles.form}
            noValidate
          >
            <Controller
              name="firstName"
              control={control}
              render={({
                field: { onChange, value },
                fieldState: { error },
              }) => {
                return (
                  <StyledInput
                    type="text"
                    id="firstName"
                    label="First Name"
                    value={value}
                    data-testid={SCHEDULE_GUEST_TEST_ID.firstname}
                    onChange={onChange}
                    error={!!error}
                    size="small"
                    variant="filled"
                    helperText={error ? error.message : null}
                    minLength={2}
                    maxLength={50}
                    aria-label={"First Name text field"}
                    InputLabelProps={{ "aria-hidden": true }}
                    sx={
                      isDesktop
                        ? { m: 1, width: "70%" }
                        : { m: 1, pr: 1, width: "100%" }
                    }
                    required
                    inputRef={firstNameRef}
                  />
                );
              }}
              rules={{
                required: t("thisFieldRequired"),
                validate: {
                  isFormat: (v) =>
                    Regex.alphabethOnly.test(v) ||
                    "Incorrect First Name format",
                  isLength: (v) =>
                    Regex.isMin2Max50Length.test(v) ||
                    "First Name should be greater than 2 characters",
                },
              }}
            />
            <Controller
              name="lastName"
              control={control}
              render={({
                field: { onChange, value },
                fieldState: { error },
              }) => {
                return (
                  <StyledInput
                    type="text"
                    id="lastName"
                    label="Last Name"
                    data-testid={SCHEDULE_GUEST_TEST_ID.lastname}
                    value={value}
                    onChange={onChange}
                    error={!!error}
                    size="small"
                    variant="filled"
                    helperText={error ? error.message : null}
                    minLength={2}
                    maxLength={50}
                    aria-label={"Last Name text field"}
                    InputLabelProps={{ "aria-hidden": true }}
                    sx={
                      isDesktop
                        ? { m: 1, width: "70%" }
                        : { m: 1, pr: 1, width: "100%" }
                    }
                    required
                    inputRef={lastNameRef}
                  />
                );
              }}
              rules={{
                required: t("thisFieldRequired"),
                validate: {
                  isFormat: (v) =>
                    Regex.alphabethOnly.test(v) || "Incorrect Last Name format",
                  isLength: (v) =>
                    Regex.isMin2Max50Length.test(v) ||
                    "Last Name should be greater than 2 characters",
                },
              }}
            />

            <Controller
              name="email"
              control={control}
              render={({
                field: { onChange, value },
                fieldState: { error },
              }) => {
                return (
                  <StyledInput
                    type="text"
                    id="email"
                    label="Email"
                    inputProps={{
                      maxLength: 50,
                    }}
                    value={value}
                    data-testid={SCHEDULE_GUEST_TEST_ID.email}
                    onChange={onChange}
                    error={!!error}
                    size="small"
                    variant="filled"
                    helperText={error ? error.message : null}
                    aria-label={"Email text field"}
                    InputLabelProps={{ "aria-hidden": true }}
                    sx={
                      isDesktop
                        ? { m: 1, width: "70%" }
                        : { m: 1, pr: 1, width: "100%" }
                    }
                    required
                    inputRef={emailRef}
                  />
                );
              }}
              rules={{
                validate: {
                  required: (value) => {
                    if (!value) {
                      if (watchedPreferredCommunication !== "phone") {
                        if (watchedPreferredCommunication === "email") {
                          return "This field is required to proceed.";
                        } else if (
                          watchedPreferredCommunication === "both" &&
                          watchedMobile
                        ) {
                          return "This field is required to proceed.";
                        } else {
                          return "Either Email or Mobile number is required to proceed";
                        }
                      }
                    }
                  },
                },
                pattern: {
                  value: Regex.emailValidation,
                  message: t("incorrectEmail"),
                },
              }}
            />
            <Controller
              name="mobile"
              control={control}
              render={({
                field: { onChange, value },
                fieldState: { error },
              }) => {
                return (
                  <StyledInput
                    type="phone"
                    id="mobile"
                    data-testid={SCHEDULE_GUEST_TEST_ID.mobilenumber}
                    label="Mobile Number"
                    value={value}
                    onChange={onChange}
                    error={!!error}
                    size="small"
                    variant="filled"
                    helperText={error ? error.message : null}
                    aria-label={"Mobile Number field"}
                    InputLabelProps={{ "aria-hidden": true }}
                    sx={
                      isDesktop
                        ? { m: 1, width: "70%" }
                        : { m: 1, pr: 1, width: "100%" }
                    }
                    required
                    inputRef={mobileRef}
                  />
                );
              }}
              rules={{
                validate: {
                  required: (value) => {
                    if (!value) {
                      if (watchedPreferredCommunication !== "email") {
                        if (watchedPreferredCommunication === "phone") {
                          return "This field is required to proceed.";
                        } else if (
                          watchedPreferredCommunication === "both" &&
                          watchedEmail
                        ) {
                          return "This field is required to proceed.";
                        } else {
                          return "Either Email or Mobile number is required to proceed";
                        }
                      }
                    }
                  },
                },
                pattern: {
                  value: Regex.isValidPhoneFormat,
                  message: t("incorrectPhone"),
                },
              }}
            />

            <Box sx={{ width: "70%" }}>
              <Controller
                name="dob"
                control={control}
                tabIndex={0}
                InputPropsLabel={{
                  "aria-label": "Date of Birth required text field",
                }}
                aria-label="Date of Birth required text field"
                render={({
                  field: { onChange, value },
                  fieldState: { error },
                }) => {
                  return (
                    <StyledInput
                      disableFuture
                      open={open}
                      onOpen={() => setOpen(true)}
                      onClose={() => {
                        setOpen(false);
                        setTimeout(() => {
                          dobRef?.current?.focus();
                        }, 1);
                      }}
                      // onClick={() => setOpen(true)}
                      aria-hidden={true}
                      tabIndex={-1}
                      type="dob"
                      id="dob"
                      InputLabel={{ "aria-hidden": false }}
                      InputLabelProps={{
                        "aria-hidden": true,
                      }}
                      InputProps={{
                        ref: dobRef,
                        tabIndex: 0,
                        "data-testid": SCHEDULE_GUEST_TEST_ID.dateofbirth,
                        "aria-label": "Date of Birth required text field",
                      }}
                      inputProps={{
                        tabIndex: -1,
                        readOnly: !isDesktop,
                        isTransparent: true,
                        "aria-label": `Password - optional -`,
                        maxLength: 50,
                      }}
                      label="Date of Birth"
                      variant="filled"
                      value={value}
                      onChange={onChange}
                      error={!!error}
                      required
                      inputRef={dobRef}
                      helperText={error ? error.message : null}
                    />
                  );
                }}
                rules={{
                  required: t("thisFieldRequired"),
                  validate: {
                    required: (value) => {
                      if (!isDOB(value)) {
                        return t("invalidDOB");
                      }
                    },
                  },
                }}
              />
            </Box>

            <div style={styles.divMargin}>
              <Controller
                name="preferredCommunication"
                control={control}
                render={({
                  field: { onChange, value },
                  fieldState: { error },
                }) => {
                  return (
                    <>
                      <RowRadioButtonsGroup
                        error={!!error}
                        value={value}
                        onChange={onChange}
                        label="Preferred mode of Communication"
                        options={options}
                        helperText={error ? error.message : null}
                        textSx={{
                          justifyContent: "space-between",
                          color: "#242526",
                          fontWeight: "600",
                          fontFamily: "Libre Franklin",
                        }}
                        sx={{
                          width: { xs: "100%", md: "56%" },
                          m: 1,
                          justifyContent: "space-between",
                          fontSize: "16px",
                          fontWeight: "600",
                          color: "black",
                        }}
                      />
                    </>
                  );
                }}
                rules={{ required: t("thisFieldRequired") }}
              />
            </div>

            {!isForMyself && <Divider />}

            {isForMyself ? (
              <>
                <Stack>
                  <Divider sx={{ mx: 1, width: { xs: "100%", md: "70%" } }} />
                  <Box
                    sx={{
                      m: "24px 8px 16px",
                      width: { xs: "100%", md: "70%" },
                    }}
                  >
                    <Typography sx={styles.optionalInfoLabel}>
                      {t("optional")}
                    </Typography>
                    <Typography sx={styles.passwordLabel}>
                      {t("passwordInfo")}
                    </Typography>
                  </Box>

                  <Controller
                    name="password"
                    control={control}
                    defaultValue=""
                    render={({
                      field: { onChange, value },
                      fieldState: { error },
                    }) => {
                      return (
                        <StyledInput
                          id="password"
                          data-testid={SCHEDULE_GUEST_TEST_ID.passwordField}
                          label={t("passwordLabel")}
                          inputProps={{
                            "aria-label": `Password - optional -`,
                            maxLength: 20,
                          }}
                          type={constants.INPUT_PASSWORD}
                          size={constants.SMALL}
                          variant={constants.FILLED}
                          value={value}
                          onChange={onChange}
                          error={!!error}
                          helperText={error ? error.message : null}
                        />
                      );
                    }}
                    rules={{
                      validate: {
                        isLength: (v) =>
                          Regex.lengthRegex.test(v) ||
                          v.length === 0 ||
                          "Password does not meet requirements",
                        isAtLeastOneNumber: (v) =>
                          Regex.numberRegex.test(v) ||
                          v.length === 0 ||
                          "Password does not meet requirements",
                        is3of4: (v) =>
                          is3of4(v) ||
                          v.length === 0 ||
                          "Password does not meet requirements",
                      },
                    }}
                  />
                  <DisclaimerText label="(Optional)" />

                  <PasswordValidator
                    validator={passwordValidator}
                    isShowValidation={isPasswordError}
                    password={watchedPassword}
                    validatePassword={validatePassword}
                  />

                  <div style={styles.registeredUsernameWrapper} tabIndex={"0"}>
                    <div>Your username will be {getRegisteredUsername()}</div>
                  </div>
                </Stack>

                <Divider sx={{ mt: 2, mx: 1 }} />
              </>
            ) : null}

            <div style={styles.divRight}>
              <Button
                data-testId="scheduleAppoinment"
                type="submit"
                variant="contained"
                sx={{
                  width: { xs: "100%", md: "222px" },
                  background: colors.primaryButton,
                  mt: 2,
                  "&:hover": {
                    backgroundColor: colors.teal,
                  },
                }}
                style={styles.continueButton}
                role={"button"}
                aria-label={t("scheduleAppoinment")}
              >
                {t("scheduleAppoinment")}
              </Button>
            </div>
          </form>
        </Stack>
      )}
    </>
  );
}
