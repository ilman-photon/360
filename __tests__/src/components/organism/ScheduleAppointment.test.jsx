import { render, fireEvent, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import ScheduleAppointment from "../../../../src/components/organisms/ScheduleAppointment/scheduleAppointment";

describe("App", () => {
    let container;
    beforeEach(() => {
      container = render(<ScheduleAppointment />);
    });

    it("ScheduleAppointment render", () => {
        expect(container).toMatchSnapshot();
    });

    it("test render", async () => {
      expect(container.getByText("formTitle")).toBeInTheDocument();

      const button1 = container.getByText("myself");
      fireEvent.click(button1);

      const button2 = container.getByText("someoneElse");
      fireEvent.click(button2);

      fireEvent.click(button1);

      waitFor(() => container.getByText("continue"));
      waitFor(() => fireEvent.click(container.getByText("continue")));
      
    });
});
