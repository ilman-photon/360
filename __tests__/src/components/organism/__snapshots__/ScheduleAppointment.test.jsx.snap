// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App ScheduleAppointment render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-1p5q5e5-MuiStack-root"
      >
        <div
          class="css-1p5q5e5-MuiStack-root"
        >
          <p
            aria-label="Who is this exam for?"
            class="MuiTypography-root MuiTypography-body1 css-fldieq-MuiTypography-root"
            tabindex="0"
          >
            formTitle
          </p>
          <div
            class="css-10wwndp-MuiStack-root"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-yr4az9-MuiButtonBase-root-MuiButton-root"
              style="border: 1px solid #003b4a; padding: 40px; border-radius: 8px; text-align: center; text-transform: none;"
              tabindex="0"
              type="button"
            >
              <h4
                aria-label="Myself option"
                class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
                style="color: rgb(0, 59, 74); font-size: 18px; font-weight: 700;"
              >
                myself
              </h4>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-yr4az9-MuiButtonBase-root-MuiButton-root"
              style="border: 1px solid #003b4a; padding: 40px; border-radius: 8px; text-align: center; text-transform: none;"
              tabindex="0"
              type="button"
            >
              <h4
                aria-label="Someone else option"
                class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
                style="color: rgb(0, 59, 74); font-size: 18px; font-weight: 700;"
              >
                someoneElse
              </h4>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <div
            style="text-align: right; width: 100%;"
          >
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-17u0idc-MuiButtonBase-root-MuiButton-root"
              disabled=""
              style="height: 40px; align-self: self-end; border-radius: 46px; text-transform: none; color: rgb(255, 255, 255);"
              tabindex="0"
              type="submit"
            >
              continue
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-1p5q5e5-MuiStack-root"
    >
      <div
        class="css-1p5q5e5-MuiStack-root"
      >
        <p
          aria-label="Who is this exam for?"
          class="MuiTypography-root MuiTypography-body1 css-fldieq-MuiTypography-root"
          tabindex="0"
        >
          formTitle
        </p>
        <div
          class="css-10wwndp-MuiStack-root"
        >
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-yr4az9-MuiButtonBase-root-MuiButton-root"
            style="border: 1px solid #003b4a; padding: 40px; border-radius: 8px; text-align: center; text-transform: none;"
            tabindex="0"
            type="button"
          >
            <h4
              aria-label="Myself option"
              class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
              style="color: rgb(0, 59, 74); font-size: 18px; font-weight: 700;"
            >
              myself
            </h4>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-yr4az9-MuiButtonBase-root-MuiButton-root"
            style="border: 1px solid #003b4a; padding: 40px; border-radius: 8px; text-align: center; text-transform: none;"
            tabindex="0"
            type="button"
          >
            <h4
              aria-label="Someone else option"
              class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
              style="color: rgb(0, 59, 74); font-size: 18px; font-weight: 700;"
            >
              someoneElse
            </h4>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <div
          style="text-align: right; width: 100%;"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-17u0idc-MuiButtonBase-root-MuiButton-root"
            disabled=""
            style="height: 40px; align-self: self-end; border-radius: 46px; text-transform: none; color: rgb(255, 255, 255);"
            tabindex="0"
            type="submit"
          >
            continue
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
